src/thunkMiddleware.lua

```diff
@@ -4,40 +4,30 @@
 	This middleware consumes the function; middleware further down the chain
 	will not receive it.
 ]]
-local function reportThunkError(errorReporter, failedAction, error_, traceback)
-	local message = ("Caught error when running thunk (%s) " ..
-		"through thunk: \n%s"):format(tostring(failedAction), tostring(error_))
-
-	errorReporter:reportErrorImmediately(message, traceback)
+local function tracebackReporter(message)
+	return debug.traceback(message)
```

Comment:
In case an non-string error gets thrown, should we `tostring` it here?

Reply:
Hmm that's a good idea, yeah